addons/pharma/functions/fnc_getBloodVolumeChange.sqf

```diff
@@ -71,11 +80,91 @@ if (!isNil {_unit getVariable [QACEGVAR(medical,ivBags),[]]}) then {
                         _ISP = _ISP + _bagChange / 2; 
                         _lossVolumeChange = _lossVolumeChange + (_bagChange / 2000);
                     } else {
-                        _ECP = _ECP + _bagChange; 
-                        _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS);
+                        { _ECP = _ECP + _bagChange; _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS); };
+                    };
+                };
+                case(_type == "Ringers Lactate"): { // TODO: talk to mike about fluid producs in trauma
+                    if (_enableFluidShift) then {
+                        _ECP = _ECP + _bagChange * 0.75; 
+                        _ISP = _ISP + _bagChange * 0.25; 
+                        _lossVolumeChange = _lossVolumeChange + (_bagChange / 2000);
+                    } else {
+                        { _ECP = _ECP + _bagChange; _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS); };
                     };
                 };
-                case(_type == "Blood"): { _ECB = _ECB + _bagChange; _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS); };
+                case(_type == "Blood"): { 
+                    _ECB = _ECB + _bagChange / 2; 
+                    _ECP = _ECP + _bagChange / 2; 
+                    _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS); 
+                };
+                case(_type == "PackedRBC"): {
+                    private _plasma = (_fluidVolume select 1);
+                    if _plasma <= 2000 then {
+                        _ECB = _ECB + _bagChange; 
+                        _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS); 
+                    } else {
+                        _ECP = _ECP + _bagChange; _lossVolumeChange = _lossVolumeChange + (_bagChange / ML_TO_LITERS);
+                    };
+                };
+            };
+        } else {
```

Comment:
My brother in christ, this runs EVERY FRAME. You are going to be adding the medication adjustment for the infusion EVERY FRAME which means tons of unnecessary duplicate medications.

Reply:
I am aware it runs every frame, which is why the medication being added has such a low TIS and TTME, 
However, there is no other way to make an infusion work without overcomplicating it to the nth degree

Reply:
My concern isnt the TIS and TTME, my concern is that we are going to have a million of the same medication which means that any function which calls the medication array, including the vitals loop, is going to have to cycle through all of those duplicate meds.

Here is a solution. When you hang one of the infusions, it starts a PFH that runs every 10(?) seconds for as long as that bag is on the line and adds the medication that way. So instead of having to add a med every frame, it only adds it every ten seconds or so, that way we only have like 8 duplicate meds instead of the hundreds that this will generate.

Reply:
The maximum in testing that i got, with a bag wide open on each limb, was 56 entries, that all get removed after a few seconds
Also, the issue with doing a PFH, its going to have to be a 1-2 second PFH, otherwise there isn't enough granularity in treatment

Reply:
Try the local PFH updating the existing value then, seems like the best bet as it would be far more lightweight and we can cycle it every two seconds or so, maintaining that granularity